icicle/Common.cmake

```diff
@@ -0,0 +1,65 @@
+function(set_env)
```

Comment:
consider putting this under a 'cmake' directory. Later we may add some other cmake scripts that do specific things and it will be easy to navigate